lib/util/storage/database/transaction_repository.dart

```diff
@@ -211,13 +211,27 @@ class MintRepository extends _TransactionRepository {
   ) async {
     try {
       assert(transaction.runtimeType == MintEntry);
-      await _store.record(transaction.transactionID).update(
+      await _store.record(transaction.blockHash).update(
             databaseClient,
             transaction.jsonMap(),
           );
     } catch (e) {
+      print('Error updating mint $e');
       return false;
     }
     return true;
   }
+
+  Future<MintEntry?> getTransaction(
+      String txHash, DatabaseClient databaseClient) async {
+    try {
+      dynamic mintInfoDbJson = await _store.record(txHash).get(databaseClient);
+
+      MintEntry valueTransferInfo = MintEntry.fromJson(mintInfoDbJson);
```

Comment:
rename this variable since it is not a valueTransfer
```suggestion
      MintEntry mintEntry = MintEntry.fromJson(mintInfoDbJson);
```